eox_nelp/utils.py

```diff
@@ -78,3 +83,39 @@ def is_valid_national_id(national_id, raise_exception=False):
         )
 
     return check_national_id
+
+
+def extract_course_id_from_string(string):
+    """This return a sub-string that matches the course_ir regex
+
+    Arguments:
+        string: This is a string that could contains a sub-string that matches with the course_id form.
+
+    Returns:
+        course_id <string>: Returns course id or an empty string.
+    """
+    matches = re.search(COURSE_ID_REGEX, string)
+
+    if matches:
+        return matches.group()
+
+    return ""
+
+
+def get_course_from_id(course_id):
+    """
+    Get Course object using the `course_id`.
+
+    Arguments:
+        course_id (str) :   ID of the course
+
+    Returns:
+        Course
+    """
+    course_key = CourseKey.from_string(course_id)
+    course_overviews = CourseOverview.get_from_ids([course_key])
+
+    if course_overviews:
+        return course_overviews[0]
+
+    raise ValueError(f"Course with id {course_id} does not exist.")
```

Comment:
Why do you prefer the raise of an exception instead a `None` return as the CourseOverview do ?
https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content/course_overviews/models.py#L415

Reply:
This is a copy of [this](https://github.com/openedx/event-routing-backends/blob/master/event_routing_backends/helpers.py#L138)